Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM python:3.9 as base
+# Perform common operations, dependency installation etc...
+
+EXPOSE 5000
+WORKDIR /app
+
+#Having to install dependencies including flask and gunicorn as hit errors without this
+RUN pip install poetry
+RUN pip install gunicorn
+RUN pip install flask
+
+COPY pyproject.toml poetry.lock ./
+
+RUN poetry install
+COPY todo_app todo_app
+
+
+FROM base as production 
+# Configure for production
```

Comment:
Something extra you could include in the Dockerfile: `ENV FLASK_ENV=production` or `ENV FLASK_ENV=development` for the two stages and `ENV FLASK_APP=todo_app/app` in the base stage.

Would save you managing them in .env files/deployment when using the Docker images. Those shouldn't need to vary at all